src/gui/Measurement.cc

```diff
@@ -50,11 +50,13 @@ void Measurement::startMeasureAngle(void)
 }
 QString Measurement::statemachine(QPoint mouse) 
 {
-  if(qglview->measure_state == MEASURE_IDLE) return "";
+  if(qglview->measure_state == MEASURE_IDLE) nullptr;
   qglview->selectPoint(mouse.x(),mouse.y());
   double ang=NAN;
   double dist=NAN;
   SelectedObject obj1, obj2, obj3;
+  Vector3d p1, p2, p3,p4, side1, side2;
+  SelectedObject ruler;
```

Comment:
It might be safer to initialize it to invalid here, in case a case becomes unhandled below:
```
SelectedObject ruler = {
  .type = SelectionType::SELECTION_INVALID
};
```